src/python/pants/backend/java/compile/javac_test.py

```diff
@@ -84,13 +84,15 @@ def rule_runner() -> RuleRunner:
 )
 
 
-JAVA_LIB_JDK12_SOURCE = dedent(
+JAVA_LIB_JDK17_SOURCE = dedent(
     """
     package org.pantsbuild.example.lib;
 
+    import javax.lang.model.SourceVersion;
+
     public class ExampleLib {
         public static String hello() {
-            return "Hello!".indent(4);
+            return "Hello " + SourceVersion.RELEASE_17 + "!";
```

Comment:
This is a more robust way to say "this release is at least this new" rather than picking some arbitrary feature of the release.